spec/e2e/has_many_spec.rb

```diff
@@ -384,4 +394,31 @@ def false_callback(_other)
       end
     end
   end
+
+  describe 'transactions' do
```

Comment:
This is just an extra test around associations and transactions in general. It will ensure that nothing accidentally happens outside of transactions.